digdag serverで実行した時に、存在するはずのsql fileが見つからない

```
2021-07-07 09:27:29.450 +0000 [INFO] (2336321@[0:debug_hori]+snow+load) io.digdag.core.agent.OperatorManager: snow>: select1.sql
2021-07-07 09:27:29.539 +0000 [ERROR] (2336321@[0:debug_hori]+snow+load) io.digdag.core.agent.OperatorManager: Task failed with unexpected error: select1.sql (No such file or directory)
java.io.FileNotFoundException: select1.sql (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at scala.io.Source$.fromFile(Source.scala:94)
	at scala.io.Source$.fromFile(Source.scala:79)
	at scala.io.Source$.fromFile(Source.scala:57)
	at dev.hiro_hori.digdag_operator_snowflake.SnowOperator.runTask(SnowOperator.scala:26)
	at io.digdag.util.BaseOperator.run(BaseOperator.java:35)
	at io.digdag.core.agent.OperatorManager.callExecutor(OperatorManager.java:365)
	at io.digdag.server.metrics.DigdagTimedMethodInterceptor.invokeMain(DigdagTimedMethodInterceptor.java:58)
	at io.digdag.server.metrics.DigdagTimedMethodInterceptor.invoke(DigdagTimedMethodInterceptor.java:31)
	at io.digdag.core.agent.OperatorManager.runWithWorkspace(OperatorManager.java:298)
	at io.digdag.server.metrics.DigdagTimedMethodInterceptor.invokeMain(DigdagTimedMethodInterceptor.java:58)
	at io.digdag.server.metrics.DigdagTimedMethodInterceptor.invoke(DigdagTimedMethodInterceptor.java:31)
	at io.digdag.core.agent.OperatorManager.lambda$runWithHeartbeat$2(OperatorManager.java:151)
	at io.digdag.core.agent.ExtractArchiveWorkspaceManager.withExtractedArchive(ExtractArchiveWorkspaceManager.java:77)
	at io.digdag.core.agent.OperatorManager.runWithHeartbeat(OperatorManager.java:149)
	at io.digdag.server.metrics.DigdagTimedMethodInterceptor.invokeMain(DigdagTimedMethodInterceptor.java:58)
	at io.digdag.server.metrics.DigdagTimedMethodInterceptor.invoke(DigdagTimedMethodInterceptor.java:31)
	at io.digdag.core.agent.OperatorManager.run(OperatorManager.java:132)
	at io.digdag.server.metrics.DigdagTimedMethodInterceptor.invokeMain(DigdagTimedMethodInterceptor.java:58)
	at io.digdag.server.metrics.DigdagTimedMethodInterceptor.invoke(DigdagTimedMethodInterceptor.java:31)
	at io.digdag.core.agent.MultiThreadAgent.lambda$null$0(MultiThreadAgent.java:132)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/src/main/scala/dev/hiro_hori/digdag_operator_snowflake/SnowOperator.scala
+++ b/src/main/scala/dev/hiro_hori/digdag_operator_snowflake/SnowOperator.scala
@@ -1,19 +1,19 @@
 package dev.hiro_hori.digdag_operator_snowflake
 
-import io.digdag.spi.{Operator, OperatorContext, OperatorFactory, TaskExecutionException, TaskResult}
+import io.digdag.spi.{Operator, OperatorContext, OperatorFactory, TaskExecutionException, TaskResult, TemplateEngine}
 import io.digdag.util.BaseOperator
 import org.slf4j.LoggerFactory
 import com.fasterxml.jackson.databind.ObjectMapper
 import io.digdag.client.config.Config
 import net.snowflake.client.jdbc.{SnowflakeDriver, SnowflakeSQLException}
 
+import java.nio.charset.StandardCharsets.UTF_8
 import java.util.Properties
 import java.sql.{Connection, DriverManager}
-import scala.io.Source
 
 
 // オペレータ本体
-class SnowOperator(_context: OperatorContext) extends BaseOperator(_context) {
+class SnowOperator(_context: OperatorContext, templateEngine: TemplateEngine) extends BaseOperator(_context) {
   private[this] val logger = LoggerFactory.getLogger(classOf[SnowOperator])
 
   override def runTask(): TaskResult = {
@@ -23,7 +23,9 @@ class SnowOperator(_context: OperatorContext) extends BaseOperator(_context) {
     val pretty = new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(config)
     logger.debug(pretty)
 
-    val source = Source.fromFile(config.get("_command", classOf[String]))
+    val command = config.get("_command", classOf[String])
+    val data = workspace.templateFile(templateEngine, command, UTF_8, config);
+
     val createTable = getOptionalParameterFromOperatorParameter(config, "create_table")
     val createOrReplaceTable = getOptionalParameterFromOperatorParameter(config, "create_or_replace_table")
     val createTableIfNotExists = getOptionalParameterFromOperatorParameter(config, "create_table_if_not_exists")
@@ -37,18 +39,17 @@ class SnowOperator(_context: OperatorContext) extends BaseOperator(_context) {
       getOptionalParameterFromOperatorParameter(config, "create_table_if_not_exists"),
       getOptionalParameterFromOperatorParameter(config, "insert_into")
     ) match {
-      case (Some(table), _, _, _) => s"CREATE TABLE $table AS " + source.mkString
-      case (_, Some(table), _, _) => s"CREATE OR REPLACE TABLE $table AS " + source.mkString
-      case (_, _, Some(table), _) => s"CREATE TABLE $table IF NOT EXISTS AS " + source.mkString
-      case (_, _, _, Some(table)) => s"INSERT INTO $table " + source.mkString
-      case _ => source.mkString
+      case (Some(table), _, _, _) => s"CREATE TABLE $table AS " + data
+      case (_, Some(table), _, _) => s"CREATE OR REPLACE TABLE $table AS " + data
+      case (_, _, Some(table), _) => s"CREATE TABLE $table IF NOT EXISTS AS " + data
+      case (_, _, _, Some(table)) => s"INSERT INTO $table " + data
+      case _ => data
     }
-    source.close()
     logger.info(sql)
 
     val conn = getConnection(
-      getConfigFromOperatorParameterOrExportedParameter(config,"host"),
-      getConfigFromOperatorParameterOrExportedParameter(config,"user"),
+      getConfigFromOperatorParameterOrExportedParameter(config, "host"),
+      getConfigFromOperatorParameterOrExportedParameter(config, "user"),
       this.context.getSecrets.getSecret("snow.password"),
       getConfigFromOperatorParameterOrExportedParameterOptional(config, "database"),
       getConfigFromOperatorParameterOrExportedParameterOptional(config, "schema"),
@@ -80,7 +81,7 @@ class SnowOperator(_context: OperatorContext) extends BaseOperator(_context) {
                      warehouse: Option[String],
                      role: Option[String],
                      unixtimeSetting: (Option[String], String),
-  ): Connection = {
+                   ): Connection = {
     DriverManager.registerDriver(
       new SnowflakeDriver()
     )
@@ -93,7 +94,7 @@ class SnowOperator(_context: OperatorContext) extends BaseOperator(_context) {
     warehouse.foreach(x => prop.put("warehouse", x))
     role.foreach(x => prop.put("role", x))
     unixtimeSetting._1.foreach(x => prop.put("$" + x, unixtimeSetting._2))
-//    logger.debug(prop.toString)
+    //    logger.debug(prop.toString)
     try {
       DriverManager.getConnection(s"jdbc:snowflake://${host}", prop)
     } catch {
@@ -122,10 +123,11 @@ class SnowOperator(_context: OperatorContext) extends BaseOperator(_context) {
 }
 
 // オペレータを生成するファクトリクラス
-class SnowOperatorFactory extends OperatorFactory {
+class SnowOperatorFactory(
+                           templateEngine: TemplateEngine,
+                         ) extends OperatorFactory {
   // ↓ これがオペレータの名前になる
   override def getType: String = "snow"
 
-  override def newOperator(context: OperatorContext): Operator =
-    new SnowOperator(context)
+  override def newOperator(context: OperatorContext): Operator = new SnowOperator(context, templateEngine)
 }

--- a/src/main/scala/dev/hiro_hori/digdag_operator_snowflake/SnowPlugin.scala
+++ b/src/main/scala/dev/hiro_hori/digdag_operator_snowflake/SnowPlugin.scala
@@ -1,13 +1,15 @@
 package dev.hiro_hori.digdag_operator_snowflake
 
 import java.util
-import io.digdag.spi.{OperatorFactory, OperatorProvider, Plugin}
+import io.digdag.spi.{OperatorFactory, OperatorProvider, Plugin, TemplateEngine}
+
 import javax.inject.Inject
 class SnowPlugin extends Plugin {
   override def getServiceProvider[T](`type`: Class[T]): Class[_ <: T] =
     if (`type` eq classOf[OperatorProvider]) classOf[SnowOperatorProvider].asSubclass(`type`) else null
 }
 
-class SnowOperatorProvider @Inject() extends OperatorProvider {
-  override def get(): util.List[OperatorFactory] = util.Arrays.asList(new SnowOperatorFactory)
+class SnowOperatorProvider @Inject()(templateEngine: TemplateEngine) extends OperatorProvider {
+
+  override def get(): util.List[OperatorFactory] = util.Arrays.asList(new SnowOperatorFactory(templateEngine))
 }

